### ISSUE TITLE
Unversioned objects and mutable versions (split out from #367)

### ISSUE THREAD

**Issue by neilsjefferies** [OCFL/spec#370]
It occurs to me that these two cases can be addressed by the same mechanism (to be decided). 

We allow an OCFL object to have a mutable head version designated by some means. Unlike something in the <code>deposit</code> directory, it must always be a valid OCFL version but can be changed by overwriting without creating a new version.

A unversioned object simply has *ony* a single mutable version. 

Implementation Notes should indicate that mutable versions should not be referenced explicitly by external systems (citation etc.) since they are not consistent. Instead such objects should be referenced at the ID level with no explicit version, or by reference to the latest stably defined vsersion.

Such versions would need more detailed Implementation Notes to cover safe updating.

Unversioned objects and objects with mutable versions should be considered a bit more "fragile" since state with repect to replication and referencing is less well defined. I think explicitly versioned persistence should remain the "gold" standard for good preservation.     

Purge redirects might also be considered as unversioned objects - it might make sense if multiple purge operations have happened.

**Comment by zimeon** [comment]
I agree that is we allow mutability, a special mutable version separate from the regular versions (`v1`, `v2`, etc.) is important so that the mutability option doesn't drag the stronger OCFL versioning guarantees down.

Not sure we need to mention or define reference mechanisms. Except for saying that objects have ids we don't do this now.

**Comment by zimeon** [comment]
https://ocfl.github.io/extensions/0005-mutable-head.html
